server/lib/pbench/test/unit/test_process_tb.py

```diff
@@ -1,67 +1,75 @@
+import os
 import logging
 import pytest
 
-from pbench import get_pbench_logger
+from pathlib import Path
+
 from pbench.process_tb import ProcessTb, Results
-from pbench.test import tarball, bad_tarball, receive_dir
+from pbench.test import tarball, bad_tarball, receive_dir, fixture_path
 
 
 class MockConfig:
-    def __init__(self):
-        self.TS = "run-1970-01-02T00:00:00.000000"
+    TS = "run-1970-01-02T00:00:00.000000"
 
     def get(self, section: str, option: str):
         pass
 
 
 class TestProcessTb:
     @pytest.fixture(autouse=True)
-    def config_and_logger(self, valid_config):
+    def logger(self, test_logger):
         # Setup the configuration and logger
-        global tmp
-        config, tmp = valid_config
-        self.logger = get_pbench_logger("pbench", config)
+        self.logger = test_logger
 
     def test_get_receive_dir_failed(self, monkeypatch):
         def mock_get(section: str, option: str):
             if option == "pbench-receive-dir-prefix":
                 return ""
+            assert False
 
         expected_error_msg = "Failed: No value for config option pbench-receive-dir-prefix in section pbench-server"
         monkeypatch.setattr(MockConfig, "get", mock_get)
         with pytest.raises(ValueError) as e:
             ProcessTb(MockConfig, self.logger)
 
-        assert str(e).endswith(expected_error_msg)
+        assert expected_error_msg in str(e)
 
     def test_get_receive_dir_value_failed(self, monkeypatch):
         def mock_get(section: str, option: str):
             if option == "pbench-receive-dir-prefix":
                 return receive_dir / "wrong-directory"
+            assert False
 
         expected_error_path = str(receive_dir / "wrong-directory-002")
         monkeypatch.setattr(MockConfig, "get", mock_get)
         with pytest.raises(Exception) as e:
             ProcessTb(MockConfig, self.logger)
 
-        assert str(e).endswith(expected_error_path)
+        assert expected_error_path in str(e)
 
     def test_process_tb_file_not_found_error(self, monkeypatch, caplog):
         """checks normal processing when tar ball is not present"""
-        tb_path = tmp / receive_dir / "bad_tarball"
+        tb_path = fixture_path / "bad_tarball"
+
+        def mock_glob(*args, **kwargs):
+            tb_path = fixture_path / "bad_tarball-002" / "bad_log.tar.xz.md5"
+            return [tb_path]
```

Comment:
This is great!

I would recommend that you have the function check its inputs, though.  I believe that the test expects it to be looking for `"**/*.tar.xz.md5"`, so if some other value gets passed in then the test probably needs to be updated, and it would be good to detect that.

Reply:
Also, I think it would be best if the function signature were an exact match for [what it's mocking](https://docs.python.org/3/library/pathlib.html#pathlib.Path.glob) (i.e., it's a member function of the `Path` class) -- I think it should look like this:
```
        def mock_glob(self, pattern)
```
(which, coincidentally, will make it easier to confirm the parameter values).

Reply:
Ditto for the other tests.

Reply:
After going through the code I realized there is no need of mocking glob because it is doing the same thing as I am trying to do by mocking. It is also creating a glob object of the same items which I am trying to pass as a list, So, I dropped it.

Reply:
I don't understand:  if the test system doesn't create a "fixtures" directory with real tarballs and MD5 files in it, then there is nothing for the real `glob()` to find, and so it won't drive all the test behaviors that we need; however, if you mock `glob()`, then you can drive the CUT exactly as needed, and we don't have to maintain external artifacts like files and directories, which makes the tests more flexible, self-contained, and simpler to deploy and run.